aws/rust-runtime/aws-inlineable/src/rds_signer.rs

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! Code related to creating signed URLs for logging in to RDS.
+//!
+//! For more information, see https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.IAMDBAuth.Connecting.html
+
+use aws_credential_types::provider::{ProvideCredentials, SharedCredentialsProvider};
+use aws_sigv4::http_request;
+use aws_sigv4::http_request::{SignableBody, SignableRequest, SigningSettings};
+use aws_sigv4::sign::v4;
+use aws_smithy_runtime_api::box_error::BoxError;
+use aws_smithy_runtime_api::client::identity::Identity;
+use aws_types::region::Region;
+use std::fmt::Debug;
+use std::time::Duration;
+
+const ACTION: &str = "connect";
+const SERVICE: &str = "rds-db";
+
+/// A signer that generates an auth token for a database.
+#[derive(Debug)]
+pub struct Signer {
+    config: SignerConfig,
+}
+
+impl Signer {
+    /// Given a `SignerConfig`, create a new RDS database login URL signer.
+    pub fn new(config: SignerConfig) -> Self {
+        Self { config }
+    }
+
+    /// Return a signed URL usable as an auth token.
+    pub async fn get_auth_token(
+        &self,
+        config: &aws_types::sdk_config::SdkConfig,
+    ) -> Result<String, BoxError> {
```

Comment:
Probably YAGNI but do we want a `String` as the result type or do we want to protect ourselves with a wrapper struct type like `struct AuthToken` that derefs to `String` or has an accessor? I'm just thinking about possible future requirements that may come along one day. Probably not worth it but just raising the question.

Reply:
I think it's worth being safe. I'll do this.